### DISCUSSION TITLE
Improve Documentation [Important] (Read Comments)

### DISCUSSION THREAD

**Discussion by HarshCasper** [HarshCasper/Rotten-Scripts#1181] [General]
## Description

Update incomplete README. As many of the Contributors might have noticed, we have a long list of Scripts along with appropriate documentation on how to launch and work with them. 

Unfortunately, a lot of Scripts have been pushed without a lot of strongest checks and reviews. We can work on improving the documentation for scripts that lack a proper information. 

Thanks to @avinal we now have a long list of Documentation that is not properly done or followed up. Here is a list of them: 

- [ ] Go/GIF_Creator/README.md
- [ ] JavaScript/Amazon_Review_Scrapper/README.md
- [ ] JavaScript/Automate_Tinder/README.md
- [ ] JavaScript/Base64_Encoder_Decoder/README.md
- [ ] JavaScript/Bulk_LinkedIn_requests/README.md
- [ ] JavaScript/Edit_Webpage/README.md
- [ ] JavaScript/Follow_Instagram/README.md
- [ ] JavaScript/Html_to_Markdown/README.md
- [ ] JavaScript/Instagram_Like/README.md
- [ ] JavaScript/Send_Emails/README.md
- [ ] JavaScript/Who_Don't_Follow_You/README.md
- [ ] Python/All_GitHub_Repos/README.md
- [ ] Python/Book_Downloader/README.md
- [ ] Python/Codechef_Contests/README.md
- [ ] Python/CRICBUZZ_scraper/README.md
- [ ] Python/cryptocurrency_converter/README.md
- [ ] Python/Currency_Converter/README.md
- [ ] Python/Download_Instagram_Posts/README.md
- [ ] Python/Download_Instagram_Videos/README.md
- [ ] Python/DSC_Event_Scrapper/README.md
- [ ] Python/Email_Scrape_From_Webpage/README.md
- [ ] Python/Encrypt_Text/README.md
- [ ] Python/Extract_Image_Metadata/README.md
- [ ] Python/File_Organiser/README.md
- [ ] Python/Find_All_Links/README.md
- [ ] Python/Flask_Directory_Initiator/README.md
- [ ] Python/Generate_Fake_Data/README.md
- [ ] Python/Get_Lyrics/README.md
- [ ] Python/Google-News-Scrapper/README.md
- [ ] Python/Heat_Your_System/README.md
- [ ] Python/HTML_to_Markdown/README.md
- [ ] Python/Image_Encoder/README.md
- [ ] Python/Image_flip_meme_templates/README.md
- [ ] Python/Image_Sketch/README.md
- [ ] Python/Image_to_Sepia/README.md
- [ ] Python/IMDb_Top_Movies/README.md
- [ ] Python/Instagram_Followers_Following_Scraper/README.md
- [ ] Python/Instagram_Followers_Following_Scraper/Using_BeautifulSoup/README.md
- [ ] Python/Internshala_Scrapper/README.md
- [ ] Python/Invisible_Cloak/README.md
- [ ] Python/JSON_to_Excel/README.md
- [ ] Python/Labelling_Binarizer_Instance/README.md
- [ ] Python/Large_File_Identifier/README.md
- [ ] Python/Lorem_Ipsum_Generator/README.md
- [ ] Python/Magic_Pen/README.md
- [ ] Python/markovmemes/README.md
- [ ] Python/Meet_Strangers/README.md
- [ ] Python/Net_Connection/README.md
- [ ] Python/OCR_To_Google_Question_Papers/README.md
- [ ] Python/Python_Library_Downloader/README.md
- [ ] Python/Query_Links_Script/README.md
- [ ] Python/Random_Password_Generator/README.md
- [ ] Python/Reddit_Wallpaper/README.md
- [ ] Python/ROT13/README.md
- [ ] Python/S3_Bucket_Downloader/README.md
- [ ] Python/Scrape_Twitter/README.md
- [ ] Python/Screen_Grabber/README.md
- [ ] Python/Search_File/README.md
- [ ] Python/Send_Emails/README.md
- [ ] Python/Shutdown_Your_System/README.md
- [ ] Python/Speech_to_Text/README.md
- [ ] Python/Story_Reader/README.md
- [ ] Python/Strong_Password_Detection/README.md
- [ ] Python/Substitution_Cypher/README.md
- [ ] Python/Test_Typing_Speed/README.md
- [ ] Python/Text_To_Speech/README.md
- [ ] Python/Times_Of_India/README.md
- [ ] Python/TSV_to_CSV/README.md
- [ ] Python/URL_Shortner/README.md
- [ ] Python/Vigenere_Cipher/README.md
- [ ] Python/Watermark_it/README.md
- [ ] Python/Weather_using_OpenWeatherMap/README.md
- [ ] Python/Website_Cloner/README.md
- [ ] Python/Who_Is_Online_On_WhatsApp/README.md
- [ ] Python/Wifi_Speed_Tester/README.md
- [ ] Python/Word_Meaning/README.md
- [ ] Python/XML_To_CSV/README.md
- [ ] Python/YouTube_Video-Audio_Download/README.md
- [ ] Python/Youtube_Video_Download/README.md
- [ ] Python/Zipper/README.md

First-Time Contributors (or even those who have landed some contributions) can take up them in batches (maybe 5-7) and raise a PR for them. As a reference take a look at our [README template](https://github.com/HarshCasper/Rotten-Scripts/blob/master/TEMPLATE_README.md) before you start modifying the Script. It would also be a good opportunity in finding some dead code, understanding how these scripts work and stepping up with future contributions.

**Comment by vybhav72954** [comment]
@HarshCasper Can i take this one, considering I am doing the styling anyways in #762 
after all the images are sorted out in the readme as mentioned in #677 

I will deal with the content of the Files

**Comment by HarshCasper** [comment]
> @HarshCasper Can i take this one, considering I am doing the styling anyways in #762
> after all the images are sorted out in the readme as mentioned in #677
> 
> I will deal with the content of the Files

Sure. Please do it in batch and let me know which batch you are working on.

**Comment by HarshCasper** [comment]
Just as Note: Anyone can work on this! Just comment the batches that you like to work on and make a PR 😉

**Comment by achalesh27022003** [comment]
Is this issue open for DWoC too?

**Comment by HarshCasper** [comment]
> Is this issue open for DWoC too?

Yeah. Go ahead

**Comment by kshittijagrawal** [comment]
Hey there!
I'd like to contribute to the project. I want to address and fix this issue. My implementation will be in python with a clean code and documentation. Kindly provide me with this issue.
I'm also a GSSOC'21 participant.

**Comment by HarshCasper** [comment]
> Hey there!
> I'd like to contribute to the project. I want to address and fix this issue. My implementation will be in python with a clean code and documentation. Kindly provide me with this issue.
> I'm also a GSSOC'21 participant.

Thanks for the interest! 

Can you take up a batch of Scripts lacking proper documentation and make a PR for? Not much, but 5-7 of them in single batches would be good.

**Comment by kshittijagrawal** [comment]
Sure, I'll try my best @HarshCasper

**Comment by BJSwaroop** [comment]
Can I work on this @HarshCasper . As I am a beginner in coding I want to Update the Documentation.

**Comment by vybhav72954** [comment]
@BJSwaroop 

> > Hey there!
> > I'd like to contribute to the project. I want to address and fix this issue. My implementation will be in python with a clean code and documentation. Kindly provide me with this issue.
> > I'm also a GSSOC'21 participant.
> 
> Thanks for the interest!
> 
> Can you take up a batch of Scripts lacking proper documentation and make a PR for? Not much, but 5-7 of them in single batches would be good.

Yes you can

**Comment by varshith-k** [comment]
@vybhav72954 I'd like to work on few documentations, Can i?

**Comment by vybhav72954** [comment]
> @vybhav72954 I'd like to work on few documentations, Can i?

Yes PLease. Open a PR as soon as you have sorted out a few.

**Comment by Charanjeev-2k2** [comment]
@vybhav72954 I would like to work on some of the documentations as a GSSoC participant, may I?

**Comment by vybhav72954** [comment]
Sure thing. You can open a PR as soon as you can sort a few README, around 4-5 (min). Please follow the proper markdown guidelines

> @vybhav72954 I would like to work on some of the documentations as a GSSoC participant, may I?

**Comment by mansijain980** [comment]
@HarshCasper  
![image](https://user-images.githubusercontent.com/63226239/113490598-c39b7200-94e8-11eb-8d88-ce2dfab22dc3.png)
![image](https://user-images.githubusercontent.com/63226239/113490615-e6c62180-94e8-11eb-97b6-895f452757cb.png)
i have a keen interest in programming python . I want to take these issues as much as i can . 
Please assign these to me .
Thanks.

**Comment by vybhav72954** [comment]
@mansijain980 The issue is open to everyone, all you need to do is, upgrade a bunch of READMEs and open a PR (Say 5-6 in one PR).

**Comment by mansijain980** [comment]
Ok thanks.

**Comment by sumitgsh** [comment]
@HarshCasper  @vybhav72954 Would love to try my hands to improve the readme. But I have one doubt like how would I figure out from the above-mentioned list which of the readme's are updated till now or is it the above list is updated one for whom no pull request is made yet.

Like suppose if someone else is working on the same readme's that I am working then I might possibly have merge conflicts

**Comment by vybhav72954** [comment]
> @HarshCasper  @vybhav72954 Would love to try my hands to improve the readme. But I have one doubt like how would I figure out from the above-mentioned list which of the readme's are updated till now or is it the above list is updated one for whom no pull request is made yet.
> 
> Like suppose if someone else is working on the same readme's that I am working then I might possibly have merge conflicts
> 

Hi sumit... Thanks for the interest... This is issue as we speak is going some changes (might be even closed). Kindly take up something else. Meanwhile we will discuss about the issue and will keep you posted as soon as we arrive at the proper goals and outcomes of this issue.

**Comment by sumitgsh** [comment]
Yah thank u..would love to know more whenever things get sorted regarding the issue..

**Comment by vybhav72954** [comment]
@HarshCasper I think it's best to close this, or maybe convert it into a discussion?

**Reply by HarshCasper** [reply]
Done @vybhav72954 

@sumitgsh You can take up 3-4 audits in a single go and send PRs for them for easy review.
